src/traces/scatter/calc.js

```diff
@@ -138,8 +249,34 @@ function calcMarkerSize(trace, serieslen) {
     }
 }
 
+/**
+ * mark the first scatter trace for each subplot
+ * note that scatter and scattergl each get their own first trace
+ * note also that I'm doing this during calc rather than supplyDefaults
+ * so I don't need to worry about transforms, but if we ever do
+ * per-trace calc this will get confused.
+ */
+function setFirstScatter(fullLayout, trace) {
+    var subplotAndType = trace.xaxis + trace.yaxis + trace.type;
+    var firstScatter = fullLayout._firstScatter;
+    if(!firstScatter[subplotAndType]) firstScatter[subplotAndType] = trace.uid;
+}
```

Comment:
In fact, `scatter_fill_corner_cases` top subplots were *also* prevented from filling to zero with `'tonexty'` because only one subplot could have the "first scatter" trace on it. This commit :hocho: `gd.firstscatter` and replaces it with one trace (uid) per subplot, attached to `fullLayout`. The stacked area mocks 🔒 this.